swap/simulations_test.go

```diff
@@ -0,0 +1,752 @@
+	nodeCount := 2
+	// create the shared backend and params
+	params, err := newSharedBackendSwaps(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// cleanup backend
+	defer params.backend.Close()
+
+	// setup the wait for mined transaction function for testing
+	cleanup := setupContractTest()
+	defer cleanup()
+
+	params.backend.cashDone = make(chan struct{}, 1)
+	defer close(params.backend.cashDone)
+	// initialize the simulation
+	sim := simulation.NewInProc(newSimServiceMap(params))
+	defer sim.Close()
+
+	log.Info("Initializing")
+
+	ctx, cancelSimRun := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancelSimRun()
+
+	_, err = sim.AddNodesAndConnectFull(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	log.Info("starting simulation...")
+
+	result := sim.Run(ctx, func(ctx context.Context, sim *simulation.Simulation) (err error) {
+		log.Info("simulation running")
+
+		p1 := sim.UpNodeIDs()[0]
+		p2 := sim.UpNodeIDs()[1]
+		p1Item, ok := sim.NodeItem(p1, bucketKeySwap)
+		if !ok {
+			return errors.New("no swap in simulation bucket")
+		}
+		ts1 := p1Item.(*testService)
+		p2Item, ok := sim.NodeItem(p2, bucketKeySwap)
+		if !ok {
+			return errors.New("no swap in simulation bucket")
+		}
+		ts2 := p2Item.(*testService)
+
+		for {
+			// let's always be nice and allow a time out to be catched
+			select {
+			case <-ctx.Done():
+				return errors.New("Timed out waiting for all swap peer connections to be established")
+			default:
+			}
+			// the node has all other peers in its peer list
+			if len(ts1.swap.peers) == 1 && len(ts2.swap.peers) == 1 {
+				break
+			}
+			// don't overheat the CPU...
+			time.Sleep(5 * time.Millisecond)
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		maxCheques := 42
+
+		p2Peer := ts1.peers[p2]
+		p1Peer := ts2.peers[p1]
+
+		for i := 0; i < maxCheques; i++ {
+			if i%2 == 0 {
+				p2Peer.Send(ctx, &testMsgBigPrice{})
+				err := waitForChequeProcessed(ts2)
+				if err != nil {
+					return err
+				}
+			} else {
+				p1Peer.Send(ctx, &testMsgBigPrice{})
+				err := waitForChequeProcessed(ts1)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		ch1, err := ts2.swap.loadLastReceivedCheque(p1)
+		if err != nil {
+			return err
+		}
+		ch2, err := ts1.swap.loadLastReceivedCheque(p2)
+		if err != nil {
+			return err
+		}
+
+		expected := uint64(maxCheques / 2 * (DefaultPaymentThreshold + 1))
+		if ch1.CumulativePayout != expected {
+			return fmt.Errorf("expected cumulative payout to be %d, but is %d", expected, ch1.CumulativePayout)
+		}
+		if ch2.CumulativePayout != expected {
+			return fmt.Errorf("expected cumulative payout to be %d, but is %d", expected, ch2.CumulativePayout)
+		}
+
+		return nil
+
+	})
+
+	if result.Error != nil {
+		t.Fatal(result.Error)
+	}
+
+	log.Info("Simulation ended")
+}
+
+// TestMultiChequeSimulation just launches two nodes, and sends multiple cheques
```

Comment:
can you please change the description of the test to include the notions you have within the test (i.e. creditor debitor and the actions which are supposed to happen between them). this will make the test more readable and easier to review

Reply:
done